### PR TITLE
ENABLE-209/223: Nightly DMS SEDS and SNS notification on SEDS DMS failure

### PR DESCRIPTION
**Pull request by sethsacher**

### CODE REVIEW

**Comment by sethsacher** [comment]
This PR will replace https://github.com/18F/cms-carts-seds/pull/528 and include both ENABLE-209 and ENABLE-223.

File: tools/dms/README.md [hunk 1]
```diff
@@ -1,21 +1,35 @@
...
\ No newline at end of file
+# Automated nightly job
+_To be run from tools/dms/serverless_
+Serverless architecture adapted from https://serverless-stack.com/
+* To run locally: npm install, serverless invoke local --function #{Function Name} --stage #{Stage Name}
+* To destroy: serverless remove
+* To deploy to AWS: serverless deploy --stage #{Stage Name}
+
+## Email notifications
+Email recipients for error notifications must be configured through the AWS SNS console
```
Comment (dwhitecl) [inline tools/dms/README.md:30]: can you give more granular step by step type of instructions for this

File: tools/dms/modules/dms/dms.tf [hunk 2]
```diff
@@ -199,3 +199,53 @@ resource "aws_security_group_rule" "postgres_ingress_from_api_postgres" {
...
   security_group_id        = data.aws_ssm_parameter.postgres_security_group.value
 }
+
+# Setup notifications for the SEDS nightly DMS job
+# TO email addresses must be configured through the AWS SNS console
+resource "aws_sns_topic" "seds-topic" {
+  name = "topic-${var.application}-${terraform.workspace}-seds"
+
+  tags = {
+    Name        = "${var.team_name} Replication Task"
```
Comment (dwhitecl) [inline tools/dms/modules/dms/dms.tf:209]: where is this getting set at?

File: tools/dms/serverless/serverless.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  iamRoleStatements:
+    - Effect: Allow
+      Action:
+        - dms:StartReplicationTask
+      Resource:
+        - ${self:custom.replicationTaskArn.${opt:stage, 'master'}}
+
+# https://www.serverless.com/framework/docs/providers/aws/events/schedule/
+# Runs daily at midnight - cron(0 0 * * ? *)
```
Comment (dwhitecl) [inline tools/dms/serverless/serverless.yml:35]: we should check with @berryd  on if nightly at midnight is the right time? I forget if we had a predetermined time or "some time at night" was agreed upon.

File: tools/dms/modules/dms/dms.tf (hunk 2 above)
Comment (sethsacher) [reply tools/dms/modules/dms/dms.tf:209]: "var" variables are set when calling this module. This happens in the directory for each workspace, in dms_task.tf

File: tools/dms/README.md (hunk 1 above)
Comment (sethsacher) [reply tools/dms/README.md]: Updated for your reivew.

File: tools/dms/serverless/serverless.yml (hunk 3 above)
Comment (sethsacher) [reply tools/dms/serverless/serverless.yml]: 5am EST was the recommendation, updated.

**Review by dwhitestratiform — APPROVED** [review]

**Pull request merged**